src/core/system/dns.hpp

```diff
@@ -3,45 +3,204 @@
 #include <string>
 #include <vector>
 #include <cstdint>
+#include <optional>
+#include <stdexcept>
+#include <sdbus-c++/sdbus-c++.h>
+#include <sysrepo.h>
+#include <libyang-cpp/DataNode.hpp>
+#include <libyang-cpp/Collection.hpp>
+#include <variant>
 
 namespace ietf::sys::dns {
 
+class Address {
```

Comment:
Maybe split into `IPv6Address` and `IPv4Address` classes? Creating the `Address` interface and then using classes for implementing it would make a cleaner design.

Reply:
The Address class determinates dynamically if it is Ipv6, or Ipv4, but if you want, we can split them into  IPv6Address and IPv4Address, and do that at runtime

Reply:
I think it would be better if we do that, yeah. This design which uses the unions was a C implementation. Now in C++ we can create a OOP approach with having the Address interface and have IPv4 and IPv6 classes implement it,